packages/ui/src/components/DataMapper/on-duplicate-datamapper.ts

```diff
@@ -0,0 +1,53 @@
+import { DataMapperMetadataService } from '../../services/datamapper-metadata.service';
+import { IMetadataApi } from '../../providers';
+import { IVisualizationNode } from '../../models';
+import { IDataMapperMetadata } from '../../models/datamapper/metadata';
+import { IClipboardCopyObject } from '../../models/visualization/clipboard';
+import { clearXsltUri, setXsltUri } from './datamapper-utils';
+
+export const onDuplicateDataMapper = async (
+  api: IMetadataApi | undefined,
+  parameters: {
+    sourceVizNode: IVisualizationNode;
+    content: IClipboardCopyObject | undefined;
+  },
+): Promise<IClipboardCopyObject | undefined> => {
+  if (!parameters.content) return parameters.content;
+
+  const stepDef = parameters.content.definition as Record<string, unknown>;
+  const newStepId = stepDef?.id as string;
+
+  if (!newStepId) return parameters.content;
+
+  if (!api) {
+    clearXsltUri(stepDef);
```

Comment:
if we follow the previous path, then it would be great to update the `clearXsltUri` to receive the `Step` interface